server/odc-core/src/main/java/com/oceanbase/odc/core/session/LogicalConnectionSession.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.core.session;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.text.DateFormat;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.FileUtils;
+
+import com.oceanbase.odc.core.datasource.DataSourceFactory;
+import com.oceanbase.odc.core.shared.constant.ConnectType;
+import com.oceanbase.odc.core.shared.constant.DialectType;
+import com.oceanbase.odc.core.shared.exception.UnsupportedException;
+import com.oceanbase.odc.core.sql.execute.AsyncJdbcExecutor;
+import com.oceanbase.odc.core.sql.execute.SyncJdbcExecutor;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @Author: Lebie
+ * @Date: 2024/8/9 14:17
+ * @Description: []
+ */
+@Slf4j
+public class LogicalConnectionSession implements ConnectionSession {
```

Comment:
is it possible to use extends current ConnectionSession for reuse some duplicated logic ?

e.g. involve an Abstract base class?

Reply:
Nice suggestion, I added an `AbstractConnectionSession` and eliminated nearly 300 lines of code.